src/controlnet_aux/processor.py

```diff
@@ -20,84 +20,117 @@
-    'content_shuffle': {'class': ContentShuffleDetector, 'checkpoint': False},
+    'shuffle': {'class': ContentShuffleDetector, 'checkpoint': False},
     'mediapipe_face': {'class': MediapipeFaceDetector, 'checkpoint': False},
     'canny': {'class': CannyDetector, 'checkpoint': False},
 }
 
-# @patrickvonplaten, I can change this so people can pass their own parameters
-# but for my use case I'm using this Dictionary
+
 MODEL_PARAMS = {
-    'hed': {'resize': False},
-    'midas': {'resize': 512},
+    'scribble_hed': {'resize': False, 'scribble': True},
+    'softedge_hed': {'resize': False, 'scribble': False},
+    'scribble_hedsafe': {'resize': False, 'scribble': True, 'safe': True},
+    'softedge_hedsafe': {'resize': False, 'scribble': False, 'safe': True},
+    'depth_midas': {'resize': 512},
     'mlsd': {'resize': False},
-    'openpose': {'resize': False, 'hand_and_face': True},
-    'pidinet': {'resize': False, 'safe': True},
-    'normalbae': {'resize': False},
-    'lineart': {'resize': False, 'coarse': True},
+    'openpose': {'resize': False, 'hand_and_face': False},
+    'openpose_face': {'resize': False, 'hand_and_face': True},
+    'openpose_faceonly': {'resize': False, 'include_body': False, 'include_face': True},
+    'openpose_full': {'resize': False, 'hand_and_face': True},
+    'scribble_pidinet': {'resize': False, 'safe': False, 'scribble': True},
+    'softedge_pidinet': {'resize': False, 'safe': False, 'scribble': False},
+    'scribble_pidsafe': {'resize': False, 'safe': True, 'scribble': True},
+    'softedge_pidsafe': {'resize': False, 'safe': True, 'scribble': False},
+    'normal_bae': {'resize': False},
+    'lineart_realistic': {'resize': False, 'coarse': False},
     'lineart_coarse': {'resize': False, 'coarse': True},
     'lineart_anime': {'resize': False},
     'canny': {'resize': False},
-    'content_shuffle': {'resize': False},
-    'zoe': {'resize': False},
+    'shuffle': {'resize': False},
+    'depth_zoe': {'resize': False},
     'mediapipe_face': {'resize': False},
 }
 
+CHOICES = f"Choices for the processor are {list(MODELS.keys())}"
+
 
 class Processor:
-    def __init__(self, processor_id: str) -> 'Processor':
+    def __init__(self, processor_id: str, params: Optional[Dict] = None) -> None:
         """Processor that can be used to process images with controlnet aux processors
 
         Args:
-            processor_id (str): processor name
-
-        Returns:
-            Processor: Processor object
+            processor_id (str): processor name, options are 'hed, midas, mlsd, openpose,
+                                pidinet, normalbae, lineart, lineart_coarse, lineart_anime,
+                                canny, content_shuffle, zoe, mediapipe_face
+            params (Optional[Dict]): parameters for the processor
         """
-        print(f"Loading {processor_id} processor")
+        LOGGER.info("Loading %s".format(processor_id))
         self.processor_id = processor_id
```

Comment:
```suggestion
        if processor_id not in MODEL_PARAMS:
            raise ValueError(f"{processor_id} is not a valid processor id. Please make sure to choose one of {', '.join(MODEL_PARAMS.keys())}".)
        self.processor_id = processor_id
```

Reply:
I do the check in the `self.load_processor(self.processor_id)` but I can move it to here if you want